### DISCUSSION TITLE
Pocketbase SQLite Extension with Custom SQL Queries

### DISCUSSION THREAD

**Discussion by noob8boi** [pocketbase/pocketbase#5117] [Q&A] [answered]
I have to perform some operations like perceptual hash matching after calculating the hamming distance (with the help of a particular SQLite extension). And I would like to know if it is possible without extending it as a framework?

**Comment by ganigeorgiev** [comment]
You can run custom queries by creating [View collections](https://pocketbase.io/docs/collections/#view-collection) but note that external extensions can't be loaded at the moment (unless you use the CGO driver but even then it will depend on the extension and its configuration whether it will work).

If your use-case relies on the extension functionality, then PocketBase is not suitable for it and you'll be better using SQLite directly.

**Reply by noob8boi** [reply]
but it depends on the hash provided by a user. So I will have to create a new view collection every time I have to get the closest hash match to the particular hash? 

I read somewhere it is possible by forking the repo and putting in the extension yourself.. tho yea not a safe approach.

**Reply by ganigeorgiev** [reply, marked as answer]
> but it depends on the hash provided by a user.

Then in this case you'll have to create a custom route and execute the raw query with Go or the JS `pb_hooks`.

But again, I don't recommend PocketBase if your use case rely on the extension.

**Reply by noob8boi** [reply]
okay thank you for the help!
